language/move-prover/boogie-backend/src/spec_translator.rs

```diff
@@ -345,24 +345,32 @@ impl<'env> SpecTranslator<'env> {
 // ============================
 
 impl<'env> SpecTranslator<'env> {
-    pub(crate) fn finalize(&self) {
+    pub(crate) fn finalize(&mut self) {
         self.translate_choice_functions();
     }
 
     /// Translate lifted functions for choice expressions.
-    fn translate_choice_functions(&self) {
+    fn translate_choice_functions(&mut self) {
         let env = self.env;
         let infos_ref = self.lifted_choice_infos.borrow();
         // need the sorting here because `lifted_choice_infos` is a hashmap while we want
         // deterministic ordering of the output. Sorting uses the `.id` field, which represents the
         // insertion order.
-        let infos_sorted = infos_ref.values().sorted_by(|v1, v2| v1.id.cmp(&v2.id));
-        for info in infos_sorted {
+        let infos_sorted_with_keys = infos_ref.iter().sorted_by(|v1, v2| v1.1.id.cmp(&v2.1.id));
```

Comment:
Once we use NodeId, we also could use BTreeMap instead of HashMap again (I think Meng used HashMap here to avoid creating an ordering on ExpData). Again, another PR.